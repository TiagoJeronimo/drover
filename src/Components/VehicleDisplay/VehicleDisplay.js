import React from 'react';
import './VehicleDisplay.style.scss';

const VehicleDisplay = ({
        vehicle,
        pricingConsumer, 
    }) => (
    <div className="vehicleDisplayContainer">
        <div className="headerInfo">
            <div className='pricingText'>
                {`Â£ ${pricingConsumer}`}
            </div>
            <div className='pricingSubtext'>
                A month
            </div>
        </div>
        <img src={vehicle.stock_image.small_image_url} alt="Vehicle" className="vehicleImage"/>
        <div className="footerInfo">
            <div className='vehicleYear'>
                {`${vehicle.year} ${vehicle.vehicle_make}`}
            </div>
            <div className='vehicleModel'>
                {`${vehicle.vehicle_model} - ${vehicle.engine_size_information}`}
            </div>
        </div>
    </div>
)


export default VehicleDisplay;