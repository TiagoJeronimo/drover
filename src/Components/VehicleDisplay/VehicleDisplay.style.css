.vehicleDisplayContainer {
  background-color: whitesmoke;
  padding: 16px;
  margin: 4px;
  flex-grow: 1;
  min-width: 25%;
  max-width: 50%; }

.vehicleImage {
  object-fit: contain;
  width: 100%;
  height: auto; }

.headerInfo {
  display: flex;
  align-items: flex-end;
  flex-direction: column; }

.pricingText {
  font-size: 22px;
  font-weight: 500; }

.pricingSubtext {
  font-size: 12px; }

.vehicleYear {
  color: #001eff;
  font-size: 14px; }

.vehicleModel {
  font-size: 18px;
  padding: 1px 0; }
