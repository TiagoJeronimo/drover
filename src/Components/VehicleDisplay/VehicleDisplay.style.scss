
.vehicleDisplay-container {
    background-color: rgb(245, 245, 245);
    padding: 16px;
    margin: 4px;
    flex-grow: 1;
    min-width: 25%;
    max-width: 50%
}

.vehicleDisplay-vehicleImage {
    object-fit: contain;
    width: 100%;
    height: auto;
}

.vehicleDisplay-headerInfo {
    display: flex;
    align-items: flex-end;
    flex-direction: column;
}

.vehicleDisplay-pricingText {
    //font-family: 'GT-Walsheim-Pro-Medium';
    font-size: 22px;
    font-weight: 500;
}

.vehicleDisplay-pricingSubtext {
    font-size: 12px;
}

.vehicleDisplay-vehicleYear {
    color: rgb(0, 30, 255);
    font-size: 14px;
}

.vehicleDisplay-vehicleModel {
    font-size: 18px;
    padding: 1px 0; 
}