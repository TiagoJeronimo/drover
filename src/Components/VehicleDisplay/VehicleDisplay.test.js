import React from 'react';
import { mount } from 'enzyme';

import VehicleDisplay from './VehicleDisplay';

it('VehicleDisplay with normal props', () => {
  const props = {
    vehicle: {
      vehicle_make: 'Citroen',
      vehicle_model: 'C1 Feel Puretech',
      year: '2017',
      engine_size_information: '1.2',
      stock_image: {
        main_image_url:
          'https://s3-eu-west-2.amazonaws.com/drover-public/uploads/stock_image/image/5661/main_C1_Puretech_Blue.png'
      }
    },
    pricingConsumer: 10
  };
  const component = <VehicleDisplay {...props} />;
  const html = mount(component).html();
  expect(html).toMatchSnapshot();
});
