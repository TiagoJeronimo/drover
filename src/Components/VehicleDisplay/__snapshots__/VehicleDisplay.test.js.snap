// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VehicleDisplay with normal props 1`] = `"<div class=\\"col-xs-12 col-sm-6 col-lg-4\\"><div class=\\"vehicleDisplay-container \\"><div class=\\"vehicleDisplay-headerInfo\\"><div class=\\"vehicleDisplay-pricingText\\">Â£10</div><div class=\\"vehicleDisplay-pricingSubtext\\">A month</div></div><img src=\\"https://s3-eu-west-2.amazonaws.com/drover-public/uploads/stock_image/image/5661/main_C1_Puretech_Blue.png\\" alt=\\"Vehicle\\" class=\\"vehicleDisplay-vehicleImage\\"><div class=\\"vehicleDisplay-footerInfo\\"><div class=\\"vehicleDisplay-vehicleYear\\">2017 Citroen</div><div class=\\"vehicleDisplay-vehicleModel\\">C1 Feel Puretech - 1.2</div></div></div></div>"`;
